x-pack/agent/pkg/agent/application/action_dispatch.go

```diff
@@ -25,9 +25,10 @@ type actionDispatcher struct {
 	log      *logger.Logger
 	handlers actionHandlers
 	def      actionHandler
+	acker    fleetAcker
 }
 
-func newActionDispatcher(log *logger.Logger, def actionHandler) (*actionDispatcher, error) {
+func newActionDispatcher(log *logger.Logger, def actionHandler, ack fleetAcker) (*actionDispatcher, error) {
```

Comment:
After looking at the code I think it would be nicer instead of injecting the `acker` it as a dependency when we create the action dispatched is to pass the acker when we call `Dispatch`, by doing this we can have more logic passed from the gateway like acking all the events in a batch triggered from the gateway.